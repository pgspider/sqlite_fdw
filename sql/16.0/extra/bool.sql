--SET log_min_messages  TO DEBUG1;
--SET client_min_messages  TO DEBUG1;
--Testcase 44:
CREATE EXTENSION sqlite_fdw;
--Testcase 45:
CREATE SERVER sqlite_svr FOREIGN DATA WRAPPER sqlite_fdw
OPTIONS (database '/tmp/sqlitefdw_test.db');

--Testcase 46:
CREATE SERVER sqlite2 FOREIGN DATA WRAPPER sqlite_fdw;

--Testcase 01:
CREATE FOREIGN TABLE "type_BOOLEAN" (i int OPTIONS (key 'true'), b bool) SERVER sqlite_svr;
--Testcase 02:
INSERT INTO "type_BOOLEAN"(i, b) VALUES (1, TRUE);
--Testcase 03:
INSERT INTO "type_BOOLEAN"(i, b) VALUES (2, FALSE);
--Testcase 04:
ALTER FOREIGN TABLE "type_BOOLEAN" ALTER COLUMN "b" TYPE text;
--Testcase 05:
CREATE FOREIGN TABLE "type_BOOLEAN+"( "i" int, "b" bool, "t" text, "l" smallint) SERVER sqlite_svr OPTIONS (table 'type_BOOLEAN+');
--Testcase 06:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (3, 'Yes');
--Testcase 07:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (4, 'YeS');
--Testcase 08:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (5, 'yes');
--Testcase 09:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (6, 'no');
--Testcase 10:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (7, 'No');
--Testcase 11:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (8, 'nO');
--Testcase 12:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (9, 'off');
--Testcase 13:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (10, 'oFf');
--Testcase 14:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (11, 'on');
--Testcase 15:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (12, 'ON');
--Testcase 16:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (13, 't');
--Testcase 17:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (14, 'T');
--Testcase 18:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (15, 'Y');
--Testcase 19:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (16, 'y');
--Testcase 20:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (17, 'F');
--Testcase 21:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (18, 'f');
--Testcase 22:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (19, 'x');
--Testcase 23:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (20, '0');
--Testcase 24:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (21, '1');
--Testcase 25:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (22, NULL);
--Testcase 26:
SELECT * FROM "type_BOOLEAN";
--Testcase 27:
ALTER FOREIGN TABLE "type_BOOLEAN" ALTER COLUMN "b" TYPE bool;
--Testcase 28:
EXPLAIN VERBOSE
SELECT * FROM "type_BOOLEAN";
--Testcase 29:
EXPLAIN VERBOSE
SELECT * FROM "type_BOOLEAN+";
--Testcase 30: ERR - invalid text affinity because not ISO:SQL text input
SELECT * FROM "type_BOOLEAN+";
--Testcase 31
DELETE FROM "type_BOOLEAN" WHERE i = 19;
--Testcase 32:
SELECT * FROM "type_BOOLEAN+";
--Testcase 33:
SELECT * FROM "type_BOOLEAN+" WHERE b IS NULL;
--Testcase 34:
SELECT * FROM "type_BOOLEAN+" WHERE b IS NOT NULL;
--Testcase 35:
SELECT * FROM "type_BOOLEAN+" WHERE b;
--Testcase 36:
SELECT * FROM "type_BOOLEAN+" WHERE NOT b;

--Testcase 47:
DROP EXTENSION sqlite_fdw CASCADE;
