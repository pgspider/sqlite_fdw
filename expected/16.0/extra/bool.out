--SET log_min_messages  TO DEBUG1;
--SET client_min_messages  TO DEBUG1;
--Testcase 44:
CREATE EXTENSION sqlite_fdw;
--Testcase 45:
CREATE SERVER sqlite_svr FOREIGN DATA WRAPPER sqlite_fdw
OPTIONS (database '/tmp/sqlitefdw_test.db');
--Testcase 46:
CREATE SERVER sqlite2 FOREIGN DATA WRAPPER sqlite_fdw;
--Testcase 01:
CREATE FOREIGN TABLE "type_BOOLEAN" (i int OPTIONS (key 'true'), b bool) SERVER sqlite_svr;
--Testcase 02:
INSERT INTO "type_BOOLEAN"(i, b) VALUES (1, TRUE);
--Testcase 03:
INSERT INTO "type_BOOLEAN"(i, b) VALUES (2, FALSE);
--Testcase 04:
CREATE FOREIGN TABLE "type_BOOLEAN+"( "i" int, "b" bool, "t" text, "l" smallint) SERVER sqlite_svr OPTIONS (table 'type_BOOLEAN+');
--Testcase 05:
INSERT INTO "type_BOOLEAN"(i, b) VALUES (3, TRUE);
--Testcase 06:
INSERT INTO "type_BOOLEAN"(i, b) VALUES (4, FALSE);
--Testcase 07:
INSERT INTO "type_BOOLEAN"(i, b) VALUES (5, true);
--Testcase 08:
INSERT INTO "type_BOOLEAN"(i, b) VALUES (6, false);
--Testcase 09:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (7, 'Yes');
--Testcase 10:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (8, 'YeS');
--Testcase 11:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (9, 'yes');
--Testcase 12:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (10, 'no');
--Testcase 13:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (11, 'No');
--Testcase 14:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (12, 'nO');
--Testcase 15:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (13, 'off');
--Testcase 16:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (14, 'oFf');
--Testcase 17:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (15, 'on');
--Testcase 18:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (16, 'ON');
--Testcase 19:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (17, 't');
--Testcase 20:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (18, 'T');
--Testcase 21:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (19, 'Y');
--Testcase 22:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (20, 'y');
--Testcase 23:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (21, 'F');
--Testcase 24:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (22, 'f');
--Testcase 25:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (23, 'x');
ERROR:  invalid input syntax for type boolean: "x"
LINE 1: INSERT INTO "type_BOOLEAN" (i, b) VALUES (23, 'x');
                                                      ^
--Testcase 26:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (24, '0');
--Testcase 27:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (25, '1');
--Testcase 28:
INSERT INTO "type_BOOLEAN" (i, b) VALUES (26, NULL);
--Testcase 29:
SELECT * FROM "type_BOOLEAN";
 i  | b 
----+---
  1 | t
  2 | f
  3 | t
  4 | f
  5 | t
  6 | f
  7 | t
  8 | t
  9 | t
 10 | f
 11 | f
 12 | f
 13 | f
 14 | f
 15 | t
 16 | t
 17 | t
 18 | t
 19 | t
 20 | t
 21 | f
 22 | f
 24 | f
 25 | t
 26 | 
(25 rows)

--Testcase 30:
ALTER FOREIGN TABLE "type_BOOLEAN" ALTER COLUMN "b" TYPE bool;
--Testcase 31:
EXPLAIN (VERBOSE, COSTS OFF)
SELECT * FROM "type_BOOLEAN";
                                QUERY PLAN                                 
---------------------------------------------------------------------------
 Foreign Scan on public."type_BOOLEAN"
   Output: i, b
   SQLite query: SELECT `i`, sqlite_fdw_bool(`b`) FROM main."type_BOOLEAN"
(3 rows)

--Testcase 32:
EXPLAIN (VERBOSE, COSTS OFF)
SELECT * FROM "type_BOOLEAN+";
                                      QUERY PLAN                                      
--------------------------------------------------------------------------------------
 Foreign Scan on public."type_BOOLEAN+"
   Output: i, b, t, l
   SQLite query: SELECT `i`, sqlite_fdw_bool(`b`), `t`, `l` FROM main."type_BOOLEAN+"
(3 rows)

--Testcase 33: ERR - invalid text affinity because not ISO:SQL text input
SELECT * FROM "type_BOOLEAN+";
 i  | b |    t    | l 
----+---+---------+---
  1 | t | integer | 1
  2 | f | integer | 1
  3 | t | integer | 1
  4 | f | integer | 1
  5 | t | integer | 1
  6 | f | integer | 1
  7 | t | integer | 1
  8 | t | integer | 1
  9 | t | integer | 1
 10 | f | integer | 1
 11 | f | integer | 1
 12 | f | integer | 1
 13 | f | integer | 1
 14 | f | integer | 1
 15 | t | integer | 1
 16 | t | integer | 1
 17 | t | integer | 1
 18 | t | integer | 1
 19 | t | integer | 1
 20 | t | integer | 1
 21 | f | integer | 1
 22 | f | integer | 1
 24 | f | integer | 1
 25 | t | integer | 1
 26 |   | null    |  
(25 rows)

--Testcase 34
DELETE FROM "type_BOOLEAN" WHERE i = 23;
--Testcase 35:
SELECT * FROM "type_BOOLEAN+";
 i  | b |    t    | l 
----+---+---------+---
  1 | t | integer | 1
  2 | f | integer | 1
  3 | t | integer | 1
  4 | f | integer | 1
  5 | t | integer | 1
  6 | f | integer | 1
  7 | t | integer | 1
  8 | t | integer | 1
  9 | t | integer | 1
 10 | f | integer | 1
 11 | f | integer | 1
 12 | f | integer | 1
 13 | f | integer | 1
 14 | f | integer | 1
 15 | t | integer | 1
 16 | t | integer | 1
 17 | t | integer | 1
 18 | t | integer | 1
 19 | t | integer | 1
 20 | t | integer | 1
 21 | f | integer | 1
 22 | f | integer | 1
 24 | f | integer | 1
 25 | t | integer | 1
 26 |   | null    |  
(25 rows)

--Testcase 36:
SELECT * FROM "type_BOOLEAN+" WHERE b IS NULL;
 i  | b |  t   | l 
----+---+------+---
 26 |   | null |  
(1 row)

--Testcase 37:
SELECT * FROM "type_BOOLEAN+" WHERE b IS NOT NULL;
 i  | b |    t    | l 
----+---+---------+---
  1 | t | integer | 1
  2 | f | integer | 1
  3 | t | integer | 1
  4 | f | integer | 1
  5 | t | integer | 1
  6 | f | integer | 1
  7 | t | integer | 1
  8 | t | integer | 1
  9 | t | integer | 1
 10 | f | integer | 1
 11 | f | integer | 1
 12 | f | integer | 1
 13 | f | integer | 1
 14 | f | integer | 1
 15 | t | integer | 1
 16 | t | integer | 1
 17 | t | integer | 1
 18 | t | integer | 1
 19 | t | integer | 1
 20 | t | integer | 1
 21 | f | integer | 1
 22 | f | integer | 1
 24 | f | integer | 1
 25 | t | integer | 1
(24 rows)

--Testcase 38:
SELECT * FROM "type_BOOLEAN+" WHERE b;
 i  | b |    t    | l 
----+---+---------+---
  1 | t | integer | 1
  3 | t | integer | 1
  5 | t | integer | 1
  7 | t | integer | 1
  8 | t | integer | 1
  9 | t | integer | 1
 15 | t | integer | 1
 16 | t | integer | 1
 17 | t | integer | 1
 18 | t | integer | 1
 19 | t | integer | 1
 20 | t | integer | 1
 25 | t | integer | 1
(13 rows)

--Testcase 39:
SELECT * FROM "type_BOOLEAN+" WHERE NOT b;
 i  | b |    t    | l 
----+---+---------+---
  2 | f | integer | 1
  4 | f | integer | 1
  6 | f | integer | 1
 10 | f | integer | 1
 11 | f | integer | 1
 12 | f | integer | 1
 13 | f | integer | 1
 14 | f | integer | 1
 21 | f | integer | 1
 22 | f | integer | 1
 24 | f | integer | 1
(11 rows)

--Testcase 40:
CREATE FOREIGN TABLE "type_BOOLEANpk" (col bool OPTIONS (key 'true')) SERVER sqlite_svr;
--Testcase 41:
INSERT INTO "type_BOOLEANpk" VALUES (TRUE);
--Testcase 42:
INSERT INTO "type_BOOLEANpk" VALUES (FALSE);
--Testcase 43: ERR - primary key
INSERT INTO "type_BOOLEANpk" VALUES (TRUE);
ERROR:  failed to execute remote SQL: rc=19 UNIQUE constraint failed: type_BOOLEANpk.col 
   sql=INSERT INTO main."type_BOOLEANpk"(`col`) VALUES (?)
--Testcase 44:
DELETE FROM "type_BOOLEANpk";
--Testcase 47:
DROP EXTENSION sqlite_fdw CASCADE;
NOTICE:  drop cascades to 5 other objects
DETAIL:  drop cascades to server sqlite_svr
drop cascades to foreign table "type_BOOLEAN"
drop cascades to foreign table "type_BOOLEAN+"
drop cascades to foreign table "type_BOOLEANpk"
drop cascades to server sqlite2
